tool/tbot/tshwrap/wrap_test.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tshwrap
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"testing"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/tool/tbot/config"
+	"github.com/gravitational/teleport/tool/tbot/identity"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+// mockRunner is a mock tsh runner
+type mockRunner struct {
+	captureErr error
+	captureOut []byte
+}
+
+func (r *mockRunner) Capture(args ...string) ([]byte, error) {
```

Comment:
I expect this mock is going to be one of the counter-arguments to my "kill the interface" comment, so I'd like to present alternatives.

1. Mock exec functions

This is similar to how folks mock `time.Now()`. You do a `var timeNow = time.Now` and use that var in your prod code. Tests swap the value for a mocked function (reset via t.Cleanup).

In your case, it's likely exec.Command, so it would be a bit of work. Other alternatives are better.

2. Detach Capture from the interface and mock it.

Ie, `var capture = func(path string, args []string) {...}` (or something like that). Same deal as above from here on.

3. Capture is mostly (only?) used for "tsh version", so create (and later mock!) a more specialized func for "tsh version". 

Possibly `var tshVersion = func(path string) (string, error) {...}`. The exec.Command invocation lives inside. Returns the JSON string on success.


These are common techniques in Go that, I believe, would push your API to be a little more idiomatic.